core/schemas/provider.go

```diff
@@ -158,4 +158,7 @@ type Provider interface {
 	TextCompletion(ctx context.Context, model, key, text string, params *ModelParameters) (*BifrostResponse, *BifrostError)
 	// ChatCompletion performs a chat completion request
 	ChatCompletion(ctx context.Context, model, key string, messages []BifrostMessage, params *ModelParameters) (*BifrostResponse, *BifrostError)
+	// Embedding generates embeddings for the given input text(s). The input can
+	// be either a single string or a slice of strings for batch embedding.
+	Embedding(ctx context.Context, model string, key string, input any, params *ModelParameters) (*BifrostResponse, *BifrostError)
```

Comment:
Would like to get agreement on this signature before building out more providers.

`input any` is not great, but limited to `string | []string`, so not too crazy.

Alternative signatures:


Option A:
```go
Embedding(ctx context.Context, model string, key string, input string, params *ModelParameters) (*BifrostResponse, *BifrostError)
BatchEmbedding(ctx context.Context, model string, key string, input []string, params *ModelParameters) (*BifrostResponse, *BifrostError)
```

Option B:
```go
type EmbeddingInput struct {
  Text string
  Texts []string
}

Embedding(ctx context.Context, model string, key string, input EmbeddingInput, params *ModelParameters) (*BifrostResponse, *BifrostError)
```

Reply:
we can go with option B, option A will lead to 2 very similar functions which will be difficult to maintain